### PR TITLE
LoggerFactory public methods throw ObjectDisposedException after dispose

### LINKED ISSUE #418: 'AddProvider' on LoggerFactory should throw ObjectDisposedException after being disposed
Simple scenario:

    var loggerFactory = new LoggerFactory();
    loggerFactory.Dispose();
    loggerFactory.AddProvider(new LoggerProviderWithUnmanagedResources());
    loggerFactory.Dispose();

The `Dispose()` on each `ILoggerProvider` is never being called a second time because the private member `_disposed` in `LoggerFactory` prevents a second call. An `ObjectDisposedException` is expected here.

Worth considering?

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolve #418 

cc @muratg @BrennanConroy

### CODE REVIEW

File: src/Microsoft.Extensions.Logging/LoggerFactory.cs [hunk 1]
```diff
@@ -19,6 +19,11 @@ public class LoggerFactory : ILoggerFactory
 
         public ILogger CreateLogger(string categoryName)
         {
+            if (_disposed)
```
Comment (BrennanConroy) [inline src/Microsoft.Extensions.Logging/LoggerFactory.cs]: Extract this out into a `CheckDisposed` method?

**Comment by BrennanConroy** [comment]
Should we throw if you try to dispose a second time also?

**Comment by natemcmaster** [comment]
No, the second dispose should just noop.

File: src/Microsoft.Extensions.Logging/LoggerFactory.cs (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.Extensions.Logging/LoggerFactory.cs]: Good thinking. I'll add.

**Comment by BrennanConroy** [comment]
:shipit:

**Comment by muratg** [comment]
:shipit:

**Pull request merged**